### ISSUE TITLE
Aggrid column filter accessibility issue

### ISSUE THREAD

**Issue by ChithiraMG** [ceolter/ag-grid#8823]
Aggrid column filtering is failing accessibility
![image](https://github.com/user-attachments/assets/8641c3dd-d62b-43e5-9c19-7c0a1a4a7504)

![image](https://github.com/user-attachments/assets/01ad710d-9485-4949-8c06-d2b17514df06)
https://www.ag-grid.com/examples/filtering/provided-filters/modules/reactFunctionalTs/

* **AG Grid version:**
*  "ag-grid-community": "^30.0.0",
    "ag-grid-enterprise": "^30.0.0",
    "ag-grid-react": "^30.0.0",

**Comment by AG-Zoheil** [comment]
Hello @ChithiraMG,
 
Thank you for reporting this.
 
This is expected behaviour because AG Grid is using a virtualised list to present the list of items in the set filter. This requires that a container element is placed between the filter element and the actual options themselves. This allows the filter to display thousands of set filter items while only using a few actual DOM elements which are reused to display different items when the set filter is scrolled. AG Grid implement custom keyboard navigation, focus handling logic and has the necessary accessibility attributes set to all the necessary elements to ensure they are announced correctly.
 
In fact the nested items displayed in the AG Grid filter are announced by the screen readers because they are correctly tagged with aria tags.
 
However, Accessibility audit tools such as axeDevTools use static DOM analysis and assume that all elements of AG Grid are rendered in the DOM, which isn't the case due to virtualization. Even though AG Grid handles keyboard navigation, focus and accessibility announcements internally and these are correct in terms of the end-user experience, the accessibility audit tools report these as issues, while in fact when the user is using the grid these aren't actual issues.
 
Please use this justification to close the issue on your side as not having any impact on the end-user experience.
 
Kind regards,
Zoheil‍
